actions/post_actions.ts

```diff
@@ -278,12 +278,13 @@ export function markPostAsUnread(post: Post, location: string) {
 
 export function deleteAndRemovePost(post: Post) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
         const {error} = await dispatch(PostActions.deletePost(post));
         if (error) {
             return {error};
         }
 
-        if (post.id === getSelectedPostId(getState() as GlobalState)) {
+        if (post.id === getSelectedPostId(state as GlobalState)) {
```

Comment:
@mattermost/web-platform this is the type casting we keep getting in many places, as the shared redux store globalstate is different from the webapp specific global state. How can we improve this.